metrics/factory.go

```diff
@@ -14,22 +14,40 @@
 
 package metrics
 
+// MetricScope defines the name and tags map associated with a metric
+type MetricScope struct {
+	Name string
+	Tags map[string]string
+}
+
+// MetricInfo defines the information associated with a metric
+type MetricInfo struct {
```

Comment:
I'd move Name/Tags here. Also can it be renamed to Options? So that the syntax doesn't stutter:
```
c := factory.Counter(metrics.Options{Name: .., Description: ...})
```